app/jobs/send_fill_invitation/session_job.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class SendFillInvitation::SessionJob < ApplicationJob
+  def perform(session_id, existing_users_emails, non_existing_users_emails, health_clinic_id = nil, intervention_id = nil)
+    session = Session.find(session_id)
+    health_clinic = HealthClinic.find_by(id: health_clinic_id)
+
+    existing_users_emails.each do |email|
+      user = User.find_by(email: email)
+      next if user && !user.email_notification
```

Comment:
I wonder, do we assume that `User.find_by(email: email)` will never return `nil`? If so, why do we check if the user exists in `next if user && !user.email_notification`? However, if we assume `user` can be `nil`, I believe we should add an extra guard `next unless user.present?`. Is it right? 🤔

Reply:
I think we do it by `next if user &&`

Reply:
if `user` is `nil` (which it seems it shouldn't as we're finding them by `existing_users_emails`, but still):
- `next if user && !user.email_notification` will not skip to the next iteration as `nil && anything` will be always false
- it will raise an exception on `user.id` in the line 12, because `user` is `nil`

If `user` cannot be `nil`, then checking its logic value in `next if user && !user.email_notification` is redundant as it will always have the logical value of `true`

Reply:
You have right